gnuradio-runtime/swig/pmt_swig.i

```diff
@@ -302,7 +319,21 @@ namespace pmt{
   bool serialize(pmt_t obj, std::streambuf &sink);
   pmt_t deserialize(std::streambuf &source);
   void dump_sizeof();
-  std::string serialize_str(pmt_t obj);
-  pmt_t deserialize_str(std::string str);
+
+  std::vector<uint8_t> _serialize_str_u8(pmt_t obj);
+  pmt_t _deserialize_str_u8(std::vector<uint8_t> py_str);
 
 } //namespace pmt
+
+%pythoncode %{
+  def serialize_str(pmt_obj):
+    import array
+    tup = _serialize_str_u8(pmt_obj)
+    return array.array('B', tup).tobytes()
```

Comment:
I'm afraid `array.array.tobytes` is Py3-only.

the base problem here being that `bytes` in Py2 gives a string, whereas it's a separate type in Py3k. 
Hm. I don't have a portable solution myself :(